src/Compiler/CompilerOptions.cpp

```diff
@@ -205,6 +205,57 @@ std::string getTargetCPUOption() {
   return (mcpu != "") ? "--mcpu=" + mcpu : "";
 }
 
+// Accel.
+static accel::Accelerator::Kind getAccelKindFromString(
+    const std::string &str, bool &found) {
+  found = true;
+  // Test each existing accelerator, returning its Kind when found.
+  APPLY_TO_ACCELERATORS(ACCEL_CL_ENUM_FROM_STRING, str);
+  // No specific accelerator found, check if we have Kind::NONE
+  if (str.compare(std::string("NONE")) == 0)
+    return accel::Accelerator::Kind::NONE;
+  // No valid accelerator found, report failure.
+  found = false;
+  return accel::Accelerator::Kind::NONE;
+}
+
+// Return 0 on success, nonzero on error.
+int setTargetAccel(const std::string &str) {
+  // Empty string means reset maccel.
+  if (str.compare(std::string("RESET")) == 0) {
+    LLVM_DEBUG(llvm::dbgs() << DEBUG_TYPE << "Set accel to empty\n");
+    maccel.clear();
+    return 0;
+  }
+  // Nonempty string means adding accelerator to current accelerator list.
+  bool found;
+  accel::Accelerator::Kind accel = getAccelKindFromString(str, found);
+  if (found) {
+    setTargetAccel(accel);
+    return 0;
+  }
+  return 1;
+}
+
+void setTargetAccel(const accel::Accelerator::Kind accel) {
+  LLVM_DEBUG(llvm::dbgs() << DEBUG_TYPE << "Set accel\"" << accel << "\"\n";);
+  // Add accel to maccel.
```

Comment:
Would remove the comment because the code is clear enough.

Reply:
that is not the intent, it is to trace what accel was added, to make sure the logic works.

Reply:
ok